cityofaustin/atd-vz-data
# Stop scroll capture on crash map and upgrade react-map-gl to v7

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/15867
https://github.com/cityofaustin/atd-data-tech/issues/15190

**Please review these PRs in this order: https://github.com/cityofaustin/atd-vz-data/pull/1398, https://github.com/cityofaustin/atd-vz-data/pull/1399, https://github.com/cityofaustin/atd-vz-data/pull/1400**

This PR:
- updates `react-map-gl` to v7 and `mapbox-gl` as well
- removes `react-map-gl-draw` and the `LocationEditMap` component that used it. This feature/component was intended to allow VZ users to edit location polygon which was never needed.
- updates the `browserslist` config in `package.json` to address a JavaScript transpile error related to Mapbox GL
   - Lots of TIL for me so [here is the background on it](https://docs.mapbox.com/mapbox-gl-js/guides/install/#targeting-transpilation-to-es6-with-browserslist) and here is [how browserslist fits into Create React App](https://create-react-app.dev/docs/supported-browsers-features/#configuring-supported-browsers). the tldr is that our config was telling [Babel](https://babeljs.io/) to make sure our app was compatible with JavaScript ES5. I think it is safe to say that all VZE users are using modern browsers that support JavaScript ES6.
   - if you want to check out the summaries of the change in browser support:
      - [current config](https://browsersl.ist/#q=%3E0.2%25%2C+not+dead%2C+not+ie+%3C%3D+9%2C+not+op_mini+all)
      - [new config in this PR that is recommended in the mapbox link above](https://browsersl.ist/#q=%3E0.2%25%2C+not+dead%2C+not+op_mini+all%2C+not+safari+%3C+10%2C+not+chrome+%3C+51%2C+not+android+%3C+5%2C+not+ie+%3C+12)

## Testing
**URL to test:** 
https://deploy-preview-1398--atd-vze-staging.netlify.app/

**Steps to test:**
1. Go to a crash page and make sure the map loads
2. See that there are zoom and fullscreen controls just like in staging and production
3. Try scrolling while hovering the mouse over the crash map. You should see a prompt to use a cooperative gesture to zoom the map in and out.
4. When the crash page loads, make sure that the map and the tile layer adjust to different heights of crash diagram to the right since they are in the same flex row. If they don't adjust correctly, you will see that the map tiles look partially loaded with some white space in the bottom portion. We don't want that!
5. Explore and drag the map around, and you can compare side-by-side with staging or production too. Zoom out and notice that you can't explore the entire planet - just within a bounding box that I stole from Moped.
6. You should expect the crash edit and location maps to **not** load. They are commented out and are updated in the next two PRs, and all of this will merge in together.

---
#### Ship list
